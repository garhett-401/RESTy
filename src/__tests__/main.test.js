import React from 'react';

//shallow just looks at the JSX that is returned
//mount checks the state/prop stuff
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

import App from '../app.js'
import Main from '../components/main.js'

describe('Should do something with stuff', () => {
  it('more description of stuff', () => {

    let app = shallow(<Main/>)
    expect(app.find('#update').exists());


  })

  it('update date on click', () => {

    
    const wrapper = mount(<App />);
    const testClick = wrapper.find('#get')
    testClick.simulate('click');
    expect(wrapper.state('method')).toEqual('GET')
  })

  it('renders correctly', () => {
    const DOM = renderer.create(<App/>).toJSON();
    expect(DOM).toMatchSnapshot();
  })
})