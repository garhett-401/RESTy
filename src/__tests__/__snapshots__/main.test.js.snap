// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should do something with stuff renders correctly 1`] = `
<div>
  <section
    class="header"
  >
    <h1>
      RESTy
    </h1>
  </section>
  <section>
    <section
      class="main"
    >
      <form
        class="form"
        onSubmit={[Function]}
      >
        <label>
          URL:
        </label>
        <input
          id="url"
          onChange={[Function]}
          placeholder="URL"
          type="text"
          value=""
        />
        <input
          id="test"
          type="submit"
          value="GO!"
        />
        <br />
        <section
          class="methods"
        >
          <ul>
            <li
              id="get"
              onClick={[Function]}
            >
              GET
            </li>
            <li
              id="post"
              onClick={[Function]}
            >
              POST
            </li>
            <li
              id="put"
              onClick={[Function]}
            >
              PUT
            </li>
            <li
              id="delete"
              onClick={[Function]}
            >
              DELETE
            </li>
          </ul>
        </section>
      </form>
    </section>
    <section
      class="urlSection"
    >
      <p
        id="update"
      >
        
         
        
      </p>
    </section>
  </section>
  <section
    class="footer"
  >
    <h3>
      Â© 2018 Code Fellows
    </h3>
  </section>
</div>
`;
